Documentation/driver-api/ancillary_bus.rst

```diff
@@ -0,0 +1,218 @@
+.. SPDX-License-Identifier: GPL-2.0-only
```

Comment:
I dont how to commit on the commit message. So commenting here

Implementation of Ancillary Bus
_[shiraz] Add ancillary bus support_

This is the initial implementation of the Ancillary Bus,
ancillary_device and ancillary_driver. The ancillary bus is
_[shiraz] You do not want to state this is an implementation. Its implicit.
Maybe something like.
Add support for ...._

a software based bus intended to support registering
ancillary_devices and ancillary_drivers and provide matching
between them and probing of the registered drivers.

The bus will support probe/remove shutdown and
suspend/resume callbacks.
_[shiraz] The bus supports ....._

Reply:
rewrote commit message using both inputs.